craft_parts/executor/part_handler.py

```diff
@@ -388,8 +389,51 @@ def _run_stage(
             files=contents.files,
             directories=contents.dirs,
             overlay_hash=overlay_hash.hex(),
+            backstage_files=backstage_contents.files,
+            backstage_directories=backstage_contents.dirs,
         )
 
+    def _migrate_to_backstage(self) -> StepContents:
+        """Migrate files from the part's export directory to the backstage directory."""
+        migratable = set(self._part.part_export_dir.rglob("*"))
+        migratable_dirs = {
+            str(path.relative_to(self._part.part_export_dir))
+            for path in migratable
+            if path.is_dir()
+        }
+        migratable_files = {
+            str(path.relative_to(self._part.part_export_dir))
+            for path in migratable
+            if path.is_file()
+        }
+
+        def pkgconfig_fixup(
+            file_path: str,
+        ) -> None:
+            if os.path.islink(file_path):
+                return
+            if not file_path.endswith(".pc"):
+                return
+            packages.fix_pkg_config(
+                prefix_prepend=self._part.backstage_dir,
+                pkg_config_file=Path(file_path),
+                prefix_trim=self._part.part_export_dir,
+            )
+            packages.fix_pkg_config(
+                prefix_prepend=self._part.backstage_dir,
+                pkg_config_file=Path(file_path),
+                prefix_trim=self._part.part_install_dir,
```

Comment:
why is this one needed? How come it references the install dir?

Reply:
If a plugin has something that gets placed in the install dir and then moved to the export dir, it might still have pkgconfig references to the install dir.